File: 0881-boats-to-save-people/0881-boats-to-save-people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        Arrays.sort(people);
        int count = 0;
        int lo = 0, hi = people.length-1;
        while(lo <= hi){
            if(people[lo]+people[hi] <= limit){
                count++;
                lo++;
                hi--;
            }
            else{
                count++;
                hi--;
            }
        }
        return count;
    }
}

File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution(object):
    def numRescueBoats(self, people, limit):
        people.sort()
        count = 0
        lo = 0
        hi = len(people) -1
        while(lo <= hi):
            if(people[lo] + people[hi] <= limit):
                count+=1
                lo+=1
                hi -=1
            else:
                count+=1
                hi-=1
        return count
        